### PR TITLE
Correctly identify bitfield constants as CONSTANT values

### PR DESCRIPTION
**Pull request by chrisr-diffblue**
Previously constant bitfield values were identified as TWO_VALUE by
default, meaning they became TOP immediately. This change makes them
be correctly identified as constants and hence have their values tracked
appropriately.

### CODE REVIEW

File: src/analyses/variable-sensitivity/variable_sensitivity_object_factory.cpp [hunk 1]
```diff
@@ -126,7 +127,7 @@ abstract_object_pointert variable_sensitivity_object_factoryt::
     return initialize_abstract_object<abstract_objectt>(
       followed_type, top, bottom, e, environment, ns);
   default:
-    assert(false);
+    UNREACHABLE;
```
Comment (chrisr-diffblue) [inline src/analyses/variable-sensitivity/variable_sensitivity_object_factory.cpp:130]: This was a bit of drive-by cleanup to keep the Linter happy...

Comment (peterschrammel) [reply src/analyses/variable-sensitivity/variable_sensitivity_object_factory.cpp:130]: Cleanups should go into a separate commit.

**Review by peterschrammel — COMMENTED** [review]
Maybe add a test that shows the effect of this change.

**Comment by martin-cs** [comment]
Including the test case they sent would be nice but I'm happy this is the right code in the right place.

**Review by thk123 — CHANGES_REQUESTED** [review]
As already pointed out this should definitely have a test that previously didn't work and now does. Otherwise looks good

**Comment by chrisr-diffblue** [comment]
Thanks for the comments folks - dumb mistake on my part not to include a regression test, now added. Its a simplified version of the original customer test case, but includes the important parts. Without the fix in this PR, the value of the `.b0` field will be TOP.

I've separated the Lint cleanup into a separate commit too, but still included in this PR to stop the Cpp Linter complaining.

File: regression/goto-analyzer/variable-sensitivity-bit-field-constants/test.desc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+CORE
+main.c
+--function main --variable --structs
+^EXIT=0$
+^SIGNAL=0$
+^bs.*\(\).*->.*\{\.byte=00001010.*@.*,.*\.b0=1.*@.*\}.*$
```
Comment (thk123) [inline regression/goto-analyzer/variable-sensitivity-bit-field-constants/test.desc]: Non-blocking, but you may prefer to:

```
__CPROVER_assert(bs.bo==1);
```

Then run with `--verify` and check for:

```
^\[main.assertion.1\] .* assertion bs.bo==1: Success$
```

As a slightly less fiddly way to validate this.

**Review by thk123 — APPROVED** [review]

File: regression/goto-analyzer/variable-sensitivity-bit-field-constants/test.desc (hunk 2 above)
Comment (martin-cs) [reply regression/goto-analyzer/variable-sensitivity-bit-field-constants/test.desc]: @thk123 you beat me to it!  Yes, that is a better way of recognising precision at a given point.

Comment (chrisr-diffblue) [reply regression/goto-analyzer/variable-sensitivity-bit-field-constants/test.desc]: Ah, cool - I like that, thanks. A bit more knowledge squirrelled away for future reference...

**Comment by chrisr-diffblue** [comment]
Regression test updated to follow @thk123 suggestion.

**Review by thk123 — APPROVED** [review]

**Comment by martin-cs** [comment]
Test case failures are in function sensitivity which I'm assured (@thk123 ?) are expected with the current branch and in the windows build, which, again, is not expected to work right now.  Thus I'm happy to merge this.
